### PR TITLE
ocamlPackages.paf: init at 0.8.0; ocamlPackages.letsencrypt: init at 0.2.4

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Dependencies of `git-paf` 3.4.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/paf/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  useDune2 = true;
+  minimumOCamlVersion = "4.08";
+
+  propagatedBuildInputs = [
+    mirage-stack
+    mirage-time
+    httpaf
+    h2
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/paf/default.nix:49]: Library `h2` is not needed for `paf` 0.0.1.

File: pkgs/development/ocaml-modules/paf/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    cohttp-lwt
+    letsencrypt
+    emile
+    ke
+    bigstringaf
+    domain-name
+    ipaddr
+    duration
+    faraday
+    ipaddr
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/paf/default.nix:61]: Library `ipaddr` is listed twice.

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/paf/default.nix]: Fixed.

File: pkgs/development/ocaml-modules/paf/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/ocaml-modules/paf/default.nix]: Thanks, fixed.

**Pull request merged**